DesignPatterns/src/main/java/decorator/Main.java

```diff
@@ -0,0 +1,12 @@
+package decorator;
+
+import java.io.*;
+
+public class Main {
+    public static void main(String[] args) throws IOException {
+        Reader reader = new UniqueReader(new InputStreamReader(new FileInputStream("test.txt")));
```

Comment:
Exception in thread "main" java.io.FileNotFoundException: test.txt (No such file or directory)